cmd/podman/common/specgen.go

```diff
@@ -771,3 +775,67 @@ func parseThrottleIOPsDevices(iopsDevices []string) (map[string]specs.LinuxThrot
 	}
 	return td, nil
 }
+
+func parseSecrets(secrets []string) ([]string, map[string]string, error) {
+	secretParseError := errors.New("error parsing secret")
+	var mount []string
+	envs := make(map[string]string)
+	for _, val := range secrets {
+		source := ""
+		secretType := ""
+		target := ""
+		split := strings.Split(val, ",")
+
+		// --secret mysecret
+		if len(split) == 1 {
+			source = val
+			mount = append(mount, source)
+			continue
+		}
+		// --secret mysecret,opt=opt
+		if !strings.Contains(split[0], "=") {
+			source = split[0]
+			split = split[1:]
+		}
+		// TODO: implement other secret options
+		for _, val := range split {
+			kv := strings.SplitN(val, "=", 2)
+			switch kv[0] {
+			case "source":
+				source = kv[1]
```

Comment:
I think there's a slight chance you could not have a second entry in the slice, perhaps:
```suggestion
				source = kv[1:]
```
to be ultrasafe?

Reply:
ditto below

Reply:
I think this is okay since strings.split takes 2 as an arg so it splits it into 2?

Reply:
Doh, yep, it's fine.  I completely messed the kv < 2 len check.